src/main/java/org/opensearch/knn/index/IndexUtil.java

```diff
@@ -59,6 +59,30 @@ public static int getFileSizeInKB(String filePath) {
         return Math.toIntExact((file.length() / BYTES_PER_KILOBYTES) + 1L); // Add one so that integer division rounds up
     }
 
+    /**
+     * Retrieves a field mapping from properties map by recursively searching through the map.
+     *
+     * @param properties A map representing properties, where each key is a property name and
+     *                   the value is either a sub-map of properties or the property value itself.
+     * @param field      The name of the field to retrieve.
+     * @return           The value of the field if found, or null if the field is not present in the map.
+     */
+    public static Object getFieldMapping(final Map<String, Object> properties, final String field) {
```

Comment:
@junqiu-lei can you please provide some explanation on how this function is able to get the nested fields.

Reply:
So, if I have a nested field, then in the parameter `field` we are just getting the deeply nested value?
example: if I have opensearch field like a.b.c.vectorfield.

then then parameter field is `vectorField` right?

Reply:
Yes, it can retrieve a field from deep nested mapping.

Reply:
Updated the method description to have more details.